src/modules/authentication/web.oidc.authentication.service.ts

```diff
@@ -16,102 +17,70 @@ const IDENTITY_SERVER_AVAILABLE_SUCCESS_STATUS_CODE: number = 200;
 export class WebOidcAuthenticationService implements IAuthenticationService {
 
   private _eventAggregator: EventAggregator;
-  private _openIdConnect: OpenIdConnect;
+  private _openIdConnect: OpenIdConnect | any;
   private _notificationService: NotificationService;
-  private _router: Router;
-  private _user: User;
-  private _logoutWindow: Window = null;
 
   constructor(eventAggregator: EventAggregator,
               notificationService: NotificationService,
-              openIdConnect: OpenIdConnect,
-              router: Router) {
+              openIdConnect: OpenIdConnect) {
     this._eventAggregator = eventAggregator;
     this._notificationService = notificationService;
     this._openIdConnect = openIdConnect;
-    this._router = router;
-
-    this._initialize();
-  }
-
-  private async _initialize(): Promise<void> {
-    const user: User = await this._openIdConnect.getUser();
-
-    const userIsNull: boolean = user === null;
-
-    this._user = userIsNull ? undefined : user;
   }
 
-  public isLoggedIn(): boolean {
-    const userIsExisting: boolean = this._user !== undefined;
+  public async isLoggedIn(authority: string): Promise<boolean> {
+    const identity: IUserIdentity = await this._getUserIdentity(authority);
 
-    return userIsExisting;
+    const userIsNotAuthorized: boolean = identity === null;
```

Comment:
you could check here for isAuthorized and then just return the variable instead of the ternary operator
```suggestion
    const userIsAuthorized: boolean = identity !== null;
```